Mathlib/Probability/Kernel/Disintegration/CdfToKernel.lean

```diff
@@ -0,0 +1,690 @@
+function `α → Measure ℝ` by taking `StieltjesFunction.measure` at each point. -/
+lemma StieltjesFunction.measurable_measure {f : α → StieltjesFunction}
+    (hf : ∀ q, Measurable fun a ↦ f a q)
+    (hf_bot : ∀ a, Tendsto (f a) atBot (𝓝 0))
+    (hf_top : ∀ a, Tendsto (f a) atTop (𝓝 1)) :
+    Measurable fun a ↦ (f a).measure := by
+  refine Measure.measurable_measure.mpr fun s hs ↦ ?_
+  have : ∀ a, IsProbabilityMeasure (f a).measure :=
+    fun a ↦ (f a).isProbabilityMeasure (hf_bot a) (hf_top a)
+  refine MeasurableSpace.induction_on_inter
+    (C := fun s ↦ Measurable fun b ↦ StieltjesFunction.measure (f b) s)
+    (borel_eq_generateFrom_Iic ℝ) isPiSystem_Iic ?_ ?_ ?_ ?_ hs
+  · simp only [measure_empty, measurable_const]
+  · rintro S ⟨u, rfl⟩
+    simp_rw [StieltjesFunction.measure_Iic (f _) (hf_bot _)]
+    simp only [sub_zero]
+    exact (hf _).ennreal_ofReal
+  · intro t ht ht_cd_meas
+    have : (fun a ↦ (f a).measure tᶜ) =
+        (fun a ↦ (f a).measure univ)
+          - fun a ↦ (f a).measure t := by
+      ext1 a
+      rw [measure_compl ht, Pi.sub_apply]
+      exact measure_ne_top _ _
+    simp_rw [this, measure_univ]
+    exact Measurable.sub measurable_const ht_cd_meas
+  · intro f hf_disj hf_meas hf_cd_meas
+    simp_rw [measure_iUnion hf_disj hf_meas]
+    exact Measurable.ennreal_tsum hf_cd_meas
+
+/-- A function `f : α × β → StieltjesFunction` with the property `IsCondKernelCDF f κ ν` gives a
+Markov kernel from `α × β` to `ℝ`, by taking for each `p : α × β` the measure defined by `f p`. -/
+noncomputable
+def IsCondKernelCDF.toKernel (f : α × β → StieltjesFunction) (hf : IsCondKernelCDF f κ ν) :
+    kernel (α × β) ℝ where
+  val p := (f p).measure
+  property := StieltjesFunction.measurable_measure hf.measurable
+    hf.tendsto_atBot_zero hf.tendsto_atTop_one
+
+lemma IsCondKernelCDF.toKernel_apply (p : α × β) : hf.toKernel f p = (f p).measure := rfl
+
+instance instIsMarkovKernel_toKernel : IsMarkovKernel (hf.toKernel f) :=
+  ⟨fun _ ↦ (f _).isProbabilityMeasure (hf.tendsto_atBot_zero _) (hf.tendsto_atTop_one _)⟩
+
+lemma IsCondKernelCDF.toKernel_Iic (p : α × β) (x : ℝ) :
+    hf.toKernel f p (Iic x) = ENNReal.ofReal (f p x) := by
+  rw [IsCondKernelCDF.toKernel_apply p, (f p).measure_Iic (hf.tendsto_atBot_zero p)]
+  simp
+
+end ToKernel
+
+section
+
+variable {f : α × β → StieltjesFunction} {hf : IsCondKernelCDF f κ ν}
+
+lemma set_lintegral_toKernel_Iic [IsFiniteKernel κ] (hf : IsCondKernelCDF f κ ν)
+    (a : α) (x : ℝ) {s : Set β} (hs : MeasurableSet s) :
+    ∫⁻ b in s, hf.toKernel f (a, b) (Iic x) ∂(ν a) = κ a (s ×ˢ Iic x) := by
+  simp_rw [IsCondKernelCDF.toKernel_Iic]
+  exact hf.set_lintegral _ hs _
+
+lemma set_lintegral_toKernel_univ [IsFiniteKernel κ] (hf : IsCondKernelCDF f κ ν)
+    (a : α) {s : Set β} (hs : MeasurableSet s) :
+    ∫⁻ b in s, hf.toKernel f (a, b) univ ∂(ν a) = κ a (s ×ˢ univ) := by
+  have : ⋃ r : ℚ, Iic (r : ℝ) = univ := by
```

Comment:
Doesn't this already exist under the name `Real.iUnion_Iic_rat`? Or maybe your plan is to delete this lemma altogether once it's not needed any more?

Reply:
Indeed it is that lemma. I was not sure of what to do with `Real.iUnion_Iic_rat`, which is currently in `Mathlib.Probability.Kernel.CondCdf`. The new file does not import it.
I figured that since it's only a few lines I could avoid it and reprove it here.

I will later refactor CondCdf to use this, so at that point I could move `Real.iUnion_Iic_rat` here, or to a better file. I added a "todo" comment to explain this.